Instant-Messenger/include/util/file_system_manager.hpp

```diff
@@ -0,0 +1,30 @@
+#include <string>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <ios>
+#include <fstream>
+#include <vector>
+
+
+
+#define MESSAGES_BASE_PATH "../../messages"
+#define FILE_EXTENSION ".csv"
+#define FILE_SEPARATOR ","
+#define PATH_SEPARATOR "/"
+
+using namespace std;
+
+using std::string;
+
+namespace filesystemmanager {
+
+using namespace message;
```

Comment:
Is there any difference from declaring using name space inside a namespace declaration? I mean, ServerMessageServer declares its using namespaces outside the namespace declaration for that specific file.

Reply:
> As a rule of thumb, using declarations should only be used after all includes in a .cpp file. There's also an item on this exact issue in the book "C++ Coding Standards" by Sutter and Alexandrescu (Item 59). Here's a quote: "But here's the common trap: Many people think that using declarations issued at namespace level (...) are safe. They are not. They are at least as dangerous, and in a subtler and more insidious way."

https://stackoverflow.com/questions/6175705/scope-of-using-declaration-within-a-namespace

Reply:
I'll keep it out of the namespace.